File: app/src/main/java/zmuzik/slidingpuzzle2/mainscreen/MainScreenView.java
package zmuzik.slidingpuzzle2.mainscreen;

import java.util.List;

import zmuzik.slidingpuzzle2.flickr.Photo;

/**
 * Created by Zbynek Muzik on 2017-03-30.
 */

public interface MainScreenView {
    void updateSavedPictures(List<String> strings);

    void setWaitingForCameraPictures();

    void updateCameraPictures(List<String> strings);

    void setWaitingForFlickrPictures();

    void updateFlickrPictures(List<Photo> strings);

}

File: app/src/main/java/zmuzik/slidingpuzzle2/mainscreen/MainScreenView.kt
package zmuzik.slidingpuzzle2.mainscreen

import zmuzik.slidingpuzzle2.flickr.Photo

/**
 * Created by Zbynek Muzik on 2017-03-30.
 */

interface MainScreenView {
    fun updateSavedPictures(pictures: List<String>)

    fun setWaitingForCameraPictures()

    fun updateCameraPictures(pictures: List<String>)

    fun setWaitingForFlickrPictures()

    fun updateFlickrPictures(photos: List<Photo>)

}
